### PR TITLE
[AP-1172] refactor E2E tests structure

### PR DESCRIPTION
**Pull request by jeet-parekh-wise**
## Problem

The E2E tests are dependent on each other in unexpected ways.

## Proposed changes

This PR refactors the structure of the tests. Makes them independent.

## Types of changes

What types of changes does your code introduce to PipelineWise?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] Description above provides context of the change
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Unit tests for changes (not needed for documentation changes)
- [ ] CI checks pass with my changes
- [x] Bumping version in `setup.py` is an individual PR and not mixed with feature or bugfix PRs
- [x] Commit message/PR title starts with `[AP-NNNN]` (if applicable. AP-NNNN = JIRA ID)
- [x] Branch name starts with `AP-NNN` (if applicable. AP-NNN = JIRA ID)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions

### CODE REVIEW

File: tests/end_to_end/target_snowflake/__init__.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+from tests.end_to_end.helpers.env import E2EEnv
+
+TEST_PROJECTS_DIR_PATH = "tests/end_to_end/test-project"
+USER_HOME = os.path.expanduser("~")
+CONFIG_DIR = os.path.join(USER_HOME, ".pipelinewise")
+
+
+class TargetSnowflake(unittest.TestCase):
+    def setUp(self) -> None:
+        super().setUp()
```
Comment (amofakhar) [inline tests/end_to_end/target_snowflake/__init__.py:16]: I am a little confused here!
why `super().setUp()` is needed here?

File: tests/end_to_end/target_snowflake/__init__.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        self.e2e_env = self.get_e2e_env()
+        self.check_snowflake_credentials_provided()
+        self.check_validate_taps()
+        self.check_import_config()
+        return None
```
Comment (amofakhar) [inline tests/end_to_end/target_snowflake/__init__.py:21]: Why `return None` is needed?

File: tests/end_to_end/target_snowflake/__init__.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    def tearDown(self) -> None:
+        return super().tearDown()
```
Comment (amofakhar) [inline tests/end_to_end/target_snowflake/__init__.py:24]: same question as `setUp`

File: tests/end_to_end/target_snowflake/__init__.py (hunk 1 above)
Comment (jeet-parekh-wise) [reply tests/end_to_end/target_snowflake/__init__.py:16]: It was generated and I didn't remove it. Let me know if you would like it to be removed.

File: tests/end_to_end/target_snowflake/__init__.py (hunk 2 above)
Comment (jeet-parekh-wise) [reply tests/end_to_end/target_snowflake/__init__.py:21]: Not "needed". I just have a habit of writing return statements explicitly. Let me know if you would like it to be removed.

File: tests/end_to_end/target_snowflake/__init__.py (hunk 3 above)
Comment (jeet-parekh-wise) [reply tests/end_to_end/target_snowflake/__init__.py:24]: Not "needed". I just have a habit of writing return statements explicitly. Let me know if you would like it to be removed.

File: tests/end_to_end/target_snowflake/__init__.py (hunk 1 above)
Comment (amofakhar) [reply tests/end_to_end/target_snowflake/__init__.py:16]: I think it would be nicer and more clean if you remove it

**Review by amofakhar — APPROVED** [review]

**Pull request merged**